### DISCUSSION TITLE
EFCore and Serilog.Settings.Configuration

### DISCUSSION THREAD

**Discussion by rmcrackan** [RehanSaeed/Serilog.Exceptions#248] [Q&A]
How would this EFCore necessity be written in json, a la [Serilog.Settings.Configuration](https://github.com/serilog/serilog-settings-configuration)

```
    .Enrich.WithExceptionDetails(new DestructuringOptionsBuilder()
        .WithDefaultDestructurers()
        .WithDestructurers(new[] { new DbUpdateExceptionDestructurer() }))
```

**Comment by RehanSaeed** [comment]
Duplicate of https://github.com/RehanSaeed/Serilog.Exceptions/issues/58

Right now, there is no way to do this and it's not clear how we could easily add support for it.

**Comment by rmcrackan** [comment]
@RehanSaeed Darn. Thanks for the reply. I actually don't like the json way; it reeks of magic strings. But I don't know of another way to have log level configurable rather than hardcoded.

**Comment by RehanSaeed** [comment]
I've used the JSON config as a switch before:

```json
"IsFooEnabled": true
```

Then in code you can read that value and enable/disable some Serilog feature.
